hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -117,6 +118,33 @@ def __init__(self, element_strategy, shape, dtype, fill, unique):
         self.element_strategy = element_strategy
         self.unique = unique
 
+        # Used by self.insert_element to check that the value can be stored
+        # in the array without e.g. overflowing.  See issue #1385.
+        if dtype.kind in (u'i', u'u'):
+            self.check_cast = lambda x: np.can_cast(x, self.dtype, 'safe')
+        elif dtype.kind == u'f' and dtype.itemsize == 2:
+            max_f2 = (2. - 2 ** -10) * 2 ** 15
+            self.check_cast = lambda x: \
+                (not np.isfinite(x)) or (-max_f2 <= x <= max_f2)
+        elif dtype.kind == u'f' and dtype.itemsize == 4:
+            max_f4 = (2. - 2 ** -23) * 2 ** 127
+            self.check_cast = lambda x: \
+                (not np.isfinite(x)) or (-max_f4 <= x <= max_f4)
```

Comment:
I feel like this code would be cleaner if these branches were merged. They only differ in the value of the `max_f*`.

Reply:
I had a go, and disagree - I could lose at most two lines and felt that all the other options were significantly uglier.  This version has some duplication, but it's clean and clear in both cases.